iModelCore/ecobjects/src/ECCustomAttribute.cpp

```diff
@@ -458,7 +458,14 @@ CustomAttributeReadStatus IECCustomAttributeContainer::ReadCustomAttributes (pug
                 {
                 // In EC3 we will fail to load the schema if any invalid custom attributes are found, for EC2 schemas we will skip the invalid attributes and continue to load the schema
                 if (containerSchema.OriginalECXmlVersionAtLeast(ECVersion::V3_0))
-                    status = CustomAttributeReadStatus::InvalidCustomAttributes;
+                    {
+                    // In EC3, skip the custom attribute if it belongs to a pruned schema
+                    Utf8String caClassName = ns.substr(0, ns.find('.'));

```

Comment:
This should be caSchemaName, not caClassName.
Also, substring is probably not a good choice here. Check how ECInstance.cpp does it around line 2600, calling SchemaKey::ParseSchemaFullName.
Looking at ECInstance.cpp line 2675 it seems we also support the ECXML_URI here, though I have never seen that in real life. The xmlns can also be missing completely if it's a local custom attribute. Try and make this check a bit more robust so it doesn't try and look-up schemas if the string format is not valid.

Reply:
Using `SchemaKey::ParseSchemaFullName()` to parse the schema name similar to how ECInstance.cpp does on line 2606 instead of `substring`. Also, we're validating the full schema name format on line 448 in ECCustomAttribute.cpp